src/controllers/dynakube/updates/version.go

```diff
@@ -51,75 +54,55 @@ func ReconcileVersions(
 		return upd, nil
 	}
 
-	var ps corev1.Secret
-	if err := cl.Get(ctx, client.ObjectKey{Name: dk.PullSecret(), Namespace: dk.Namespace}, &ps); err != nil {
-		return upd, errors.WithMessage(err, "failed to get image pull secret")
-	}
-
-	auths, err := dtversion.ParseDockerAuthsFromSecret(&ps)
+	caCertPath := path.Join(TmpCAPath, TmpCAName)
+	dockerConfig, err := dockerconfig.NewDockerConfig(ctx, kubeClient, *dkState.Instance)
 	if err != nil {
-		return upd, errors.WithMessage(err, "failed to get Dockerconfig for pull secret")
+		return upd, err
 	}
-
-	dockerCfg := dtversion.DockerConfig{Auths: auths, SkipCertCheck: dk.Spec.SkipCertCheck}
 	if dk.Spec.TrustedCAs != "" {
-		dockerCfg.UseTrustedCerts = saveCustomCAs(cl, *dk)
+		_ = os.MkdirAll(TmpCAPath, 0755)
+		dockerConfig.SaveCustomCAs(ctx, kubeClient, *dkState.Instance, caCertPath)
+		if err != nil {
+			return upd, err
+		}
 		defer func() {
-			_ = os.Remove(path.Join(dtversion.TmpCAPath, dtversion.TmpCAName))
+			_ = os.Remove(TmpCAPath)
 		}()
 	}
 	upd = true // updateImageVersion() always updates the status
 
 	if needsActiveGateUpdate {
-		if err := updateImageVersion(dkState, dk.ActiveGateImage(), &dk.Status.ActiveGate.VersionStatus, &dockerCfg, verProvider, true); err != nil {
+		if err := updateImageVersion(dkState, dk.ActiveGateImage(), &dk.Status.ActiveGate.VersionStatus, dockerConfig, verProvider, true); err != nil {
```

Comment:
This function must be refactored somehow. Not only is it called with six parameters, the parameters are almost the same. Also the `allowDowngrades` flag seems weird. Additionally, these condtionals could be moved into a function.

Reply:
The updates package should be refactored in another PR